pkgs/by-name/em/emmylua_ls/package.nix

```
@@ -0,0 +1,41 @@
+{
+  fetchFromGitHub,
+  lib,
+  rustPlatform,
+  versionCheckHook,
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "emmylua_ls";
+
+  version = "0.10.0";
+
+  src = fetchFromGitHub {
+    owner = "EmmyLuaLs";
+    repo = "emmylua-analyzer-rust";
+    tag = finalAttrs.version;
+    hash = "sha256-Fvg3G0C/YECDEWZ4mDC5b8qocWvyDJ9KdLYNtwIu0+I=";
+  };
+
+  buildAndTestSubdir = "crates/emmylua_ls";
+
+  cargoHash = "sha256-MIGYx1qMxsCCq3QkFeOuKbM4w/sJ2K0T+SRIDJQjf/8=";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/${finalAttrs.meta.mainProgram}";
```

```suggestion
  versionCheckProgram = "${placeholder "out"}/bin/emmylua_ls";
```
Usually we prefer hard-coding the binary name here.